csharp/src/Apache.Arrow.Compression/Lz4CompressionCodec.cs

```diff
@@ -56,6 +61,21 @@ public void Compress(ReadOnlyMemory<byte> source, Stream destination)
             encoder.WriteManyBytes(source.Span);
         }
 
+        public bool TryCompress(ReadOnlyMemory<byte> source, Memory<byte> destination, out int bytesWritten)
+        {
+            using var memoryStream = new MemoryStream(destination.Length);
```

Comment:
Is this not a regression in behaviour by creating a new `MemoryStream` here for each `TryCompress` call rather than reusing the stream from `ArrowRecordBatchFlatBufferBuilder._compressionStream`?